### PR TITLE
fix(kit): possible type error reduce of empty array with no initial value

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Reduce_of_empty_array_with_no_initial_value

### CODE REVIEW

File: projects/kit/components/input-time/input-time.component.ts [hunk 1]
```diff
@@ -296,11 +296,17 @@ export class TuiInputTimeComponent
...
     private findNearestTimeFromItems(value: TuiTime): TuiTime | null {
-        return this.items.reduce((previous, current) =>
-            Math.abs(current.toAbsoluteMilliseconds() - value.toAbsoluteMilliseconds()) <
-            Math.abs(previous.toAbsoluteMilliseconds() - value.toAbsoluteMilliseconds())
-                ? current
-                : previous,
+        return this.items.reduce(
+            (previous, current) =>
+                Math.abs(
+                    current.toAbsoluteMilliseconds() - value.toAbsoluteMilliseconds(),
```
Comment (waterplea) [inline projects/kit/components/input-time/input-time.component.ts:302]: You can use `valueOf()`, it's shorter.

File: projects/addon-charts/components/bar-chart/bar-chart.component.ts [hunk 2]
```diff
@@ -94,6 +94,7 @@ export class TuiBarChartComponent {
     private getMax(values: ReadonlyArray<readonly number[]>, collapsed: boolean): number {
         return collapsed
             ? Math.max(
+                  // eslint-disable-next-line no-restricted-syntax
```
Comment (waterplea) [inline projects/addon-charts/components/bar-chart/bar-chart.component.ts:97]: What happens if you add default `0` here?

**Review by waterplea — APPROVED** [review]

File: projects/addon-charts/components/bar-chart/bar-chart.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-charts/components/bar-chart/bar-chart.component.ts:97]: image

Comment (splincode) [reply projects/addon-charts/components/bar-chart/bar-chart.component.ts:97]: if I add `[]`
image

tests are failed
image

File: projects/kit/components/input-time/input-time.component.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/input-time/input-time.component.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**